linkis-commons/linkis-scheduler/src/main/scala/org/apache/linkis/scheduler/queue/fifoqueue/FIFOUserConsumer.scala

```diff
@@ -188,6 +190,19 @@ class FIFOUserConsumer(
     runningJobs(index) = job
   }
 
+  protected def scanAllRetryJobsAndRemove(): util.List[Job] = {
+    val jobs = new util.ArrayList[Job]()
+    for (index <- runningJobs.indices) {
+      val job = runningJobs(index)
+      if (job != null && job.isJobCanRetry) {
+        jobs.add(job)
+        runningJobs(index) = null
```

Comment:
When a service is marked offline, it waits for the running job to complete instead of stop. The retryjob can be understood as a new task, so failover can be performed in advance. and reduce the waiting time for going offline.
(当服务标记为offline,是会等待running 任务完成，而不是stop. 等待重试任务可以理解为新任务，所以可以提前进行failover. 减少等待下线时长)

Reply:
I see the failover task will first kill the ec, and then submit a new job? is this means the running job will running again, when the failover server get the task?

Reply:
Correction: The retry job is special and is not really running(更正：retry job比较特殊，并且不是真正意义上的running)
It is better to let other nodes execute than to  wait for the task to finish normally（与其等待再次被提交执行，不如交由其他节点执行）